Std/Data/Array/Lemmas.lean

```diff
@@ -724,3 +724,182 @@ theorem all_eq_true_iff_forall_mem {l : Array α} : l.all p ↔ ∀ x, x ∈ l 
 /-! ### erase -/
 
 @[simp] proof_wanted erase_data [BEq α] {l : Array α} {a : α} : (l.erase a).data = l.data.erase a
+
+/-! ### swap -/
+
+@[simp] theorem swap_def_jeqi {a: Array α} {i j : Nat}
+    (_ : i < a.size := by trivial) (_ : j < a.size := by trivial)
+    (_ : j < (a.swap ⟨i, ‹_›⟩ ⟨j, ‹_›⟩).size := by trivial) : (a.swap ⟨i, _⟩ ⟨j, _⟩)[j] = a[i] :=
```

Comment:
This shouldn't be a separate argument, it follows from the previous one.

Reply:
Lemma should be called `get_swap_right`, I think.

Reply:
will provide `Fin` params instad of `Nat` for simplicity